src/datepicker/datepicker.js

```diff
@@ -161,9 +161,7 @@ angular.module('ui.bootstrap.datepicker', ['ui.bootstrap.dateparser', 'ui.bootst
   $scope.keys = { 13:'enter', 32:'space', 33:'pageup', 34:'pagedown', 35:'end', 36:'home', 37:'left', 38:'up', 39:'right', 40:'down' };
 
   var focusElement = function() {
-    $timeout(function() {
```

Comment:
@chrisirhc This PR does not include #3705 suggested changes.  I moved the $timeout wrapping the focus() call to the isOpen watch and wrapped the datepicker.focus broadcast with it.  This was necessary as the datepicker popup now needs a digest cycle to render with ng-if and will provide a better indication of when the datepicker is ready for focus.  Otherwise self.element[0] was not in the DOM yet.

I would like to move the document click binding into that same timeout, but will submit another PR for that.